torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -57,10 +79,15 @@ void DistAutogradContainer::releaseContext(int64_t context_id) {
       "Could not find autograd context with id: ",
       context_id);
   autograd_context_.erase(context_id);
+
+  if (current_context_id_ == context_id) {
+    // Reset the thread_local current context id, since it is no longer valid.
+    current_context_id_ = -1;
+  }
```

Comment:
Not part of this PR per se, but do we have a C++-side RAII guard for allocating and releasing thread local context? It seems like a logical thing that should be part of the C++ API.

Reply:
C++ API is non-goal for now, but let's add a TODO here in case we forgot.